README.md

```diff
@@ -1,298 +1,433 @@
+
+- A single value
 
-`fi.reduce(collection, callback, acc)`
+Behavior:
 
-Reduce boils down a **collection** of values into a single value. **Acc**
-(short for accumulator) starts as the initial state of the reduction, and with
-each successive step it should be accumulate the return value of **callback**.
-The callback is passed three arguments: the acc, the current value in our
-iteration (the element in the array), and finally a reference to the entire
+Reduce iterates through a `collection` of values and boils it down into a single
+value. `acc` (short for accumulator) starts at the value that's passed in as an
+argument, and with each successive step is updated to the return value of
+`callback`. If a start value is not passed to `myReduce`, the first value in
+the collection should be used as the start value.
+
+The `callback` is passed three arguments: the current value of `acc`, the
+current element/value in our iteration, and a reference to the entire
 collection.
 
+**Hint:** For the case when a start value is not passed in as an argument, think
+about how you'll need to adjust your function to account for the fact that the
+first element of the collection has already been accounted for.
+
+Example function calls:
+
 ```javascript
-var sum = fi.reduce([1, 2, 3], function(acc, val, collection) { return acc + val; }, 0);
+myReduce([1, 2, 3], function(acc, val, collection) { return acc + val; }, 10);
+=> 16
+
+myReduce({one: 1, two: 2, three: 3}, function(acc, val, collection) { return acc + val; });
 => 6
 ```
 
-**fi.find**
+### myFind
+
+`myFind(collection, predicate)`
+
+Argument(s):
 
-`fi.find(collection, predicate)`
+- a collection (either an object or an array)
+- a predicate (a test that returns `true` or `false`)
 
-Looks through each value in the **collection**, returning the first one that
-passes a truth test (**predicate**), or undefined if no value passes the test.
-The function returns as soon as it finds an acceptable element, and doesn't
-traverse the entire collection.
+Return value:
+
+- A single value
+
+Behavior:
+
+Looks through each value in the `collection`, returning the first one that
+passes a truth test (`predicate`) or undefined if no value passes the test. The
+function should return as soon as it finds an acceptable element, without
+traversing the rest of the collection.
+
+Example function calls:
 
 ```javascript
-var even = fi.find([1, 2, 3, 4, 5, 6], function(num){ return num % 2 == 0; });
+myFind([1, 2, 3, 4, 5, 6], function(num){ return num % 2 == 0; });
 => 2
+
+myFind({one: 1, three: 3, four: 4, six: 6}, function(num){ return num % 2 == 0; });
+=> 4
 ```
 
-**fi.filter**
+### myFilter
+
+`myFilter(collection, predicate)`
 
-`fi.filter(collection, predicate)`
+Argument(s):
 
-Looks through each value in the **collection**, returning an array of all the
-values that pass a truth test (**predicate**).
+- a collection (either an object or an array)
+- a predicate (a test that returns `true` or `false`)
```

Comment:
Suggested change:

- a predicate (a callback function that returns `true` or `false`)

Reply:
Done

Reply:
Done